script_based_workflows.Rmd

```diff
@@ -0,0 +1,201 @@
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+```{r, message = FALSE, warning = FALSE, echo = FALSE}
+library(drake)
+library(glue)
+library(purrr)
+library(rlang)
+library(tidyverse)
+invisible(drake_example("script-based-workflows", overwrite = TRUE))
+tmp <- file.copy(
+  list.files("script-based-workflows/R/",
+             pattern = "*.R", full.names = TRUE),
+  ".",
+  recursive = TRUE)
+tmp <- file.copy("script-based-workflows/raw_data.xlsx", ".")
+tmp <- file.copy("script-based-workflows/report.Rmd", ".")
+dir.create("data")
+```
+
+`drake` has been built with the firm belief that the best workflows rely on approaching problems from a function perspective. This idea is supported in "Good enough practices in scientific computing", that approaching your project by writing modular code increases the end product.
+
+> The core realization in these practices is that being readable, reusable, and testable are all side effects of writing modular code, i.e., of building programs out of short, single-purpose functions with clearly-defined inputs and outputs 
+> 2017 Wilson et al.
+
+That being said, not all projects and workflows are written using this mantra and for others the lift would be too great to re-factor to a purely functional approach. 
+
+## Traditional and Legacy Workflows
+
+A common approach to creating workflows is to write steps of the analysis into scripts that perform specific tasks. By giving these scripts meaningful names, e.g. 01_data.R, 02_munge.R, 03_analysis.R,  one can surmise both their contents and the order in which these scripts should be run.
+
+It was acknowledged in the (What we left out)[https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510#sec027] section that build tools to manage projects when code gets out of date are ideal, but can be substituted with a simple master script called `make.R` that reruns each of the scripts in order.
+
+```{r,eval=FALSE}
+
+# Preprocessing steps
+source("01_data.R")
+source("02_munge.R")
+source("03_histogram.R")
+source("04_regression.R")
+
+# Calls rmarkdown::render on report.Rmd
+source("05_report.R")
+```
+
+The argument being that while some code may be reran, for small projects this likely is not going to be a bottleneck. 
+
+## Overcoming Technical Debt
+
+The argument against the workflow manager as above is the whole reason `drake` was developed: For non-trivial or long running processes, simply rerunning all the steps is unfeasible.
+
+> With the `code_to_function()` we can now move these encumbered workflows into the managed plan ecosystem of `drake`.
+
+`code_to_function()` accepts the path to either an R or Rmd script and will generate a function that contains the contents of the script you provided as a function.
+
+
+```{r}
+
+load_data <- code_to_function("01_data.R")
+
+```
+
+By turning these scripts into functions, they can be added into a `drake` plan for workflow management.
+
+
+## Dependencies
```

Comment:
People often think of dependencies as R packages and system requirements. When we talk about `drake`, and about declarative workflows in general, we use that word differently. In this context, dependencies are targets, functions, files, etc., anything that could invalidate downstream results when changed. So I think it is worth touching on what we mean by "dependency" and why it matters, at least as far as being able to skip up-to-date computation. I should probably go back earlier in the manual to make this clear in other places as well.

Reply:
I added a little more on what a dependency is, but part of me assumes that people have leafed through the manual to describe what a `dependency` is. We can add more details if you think it would be helpful though!